### ISSUE TITLE
How to get network to work?

### ISSUE THREAD

**Issue by cleanremote** [copy/v86#392]
![Screen Shot 2021-01-19 at 11 40 51 AM](https://user-images.githubusercontent.com/27508072/105031510-b0364a00-5a55-11eb-909e-6381cc60a8c6.png)

**Comment by copy** [comment]
Probably in a bug in ne2k.js.

**Comment by gitrats07** [comment]
wait how did you boot windows 7?

**Comment by hello-smile6** [comment]
How did you get Windows 7 to work, @cleanremote?

**Comment by ayunami2000** [comment]
can you send the 7.img file that you used? (upload it to google drive or mediafire or github)

**Comment by BenNottelling** [comment]
AFAIK Vista and newer don't have support for the NE2K, buuuut there are two possible solutions that may or may not work:
1) Use the NE2K driver from a Windows Longhorn build: https://www.betaarchive.com/forum/viewtopic.php?t=38054
2) Use the NE2K driver from ReactOS. [ReactOS NE2K.zip](https://github.com/copy/v86/files/8230099/ReactOS.NE2K.zip) (Compiled today off of their current master branch)


I tried option #2 personally.

Needed to tell Windows to install the driver (not signed), but after quite a bit of thinking it did indeed install and it said that it was working, but I still didn't see a connection inside of Windows. Windows doesn't even seem to show it as a network adapter in control panel.

![allow](https://user-images.githubusercontent.com/17304943/157826200-75735b66-3ba2-4e1b-9a02-85564146d47b.png)
![image](https://user-images.githubusercontent.com/17304943/157826317-731f346b-8125-4ab7-ac93-78c92276d7d3.png)
![image](https://user-images.githubusercontent.com/17304943/157826371-512fd095-b025-4c88-aaf3-332fcd2b439f.png)
![image](https://user-images.githubusercontent.com/17304943/157827766-3e40ed06-8ba5-4b18-b5e3-8b459befc1d7.png)


Another oddity is that Window seems to PCI devices are removable?
@copy Any idea what's going on with this? 
![image](https://user-images.githubusercontent.com/17304943/157828016-1c49a490-cfb6-4e02-a7ac-b79ffc7d236f.png)

I tried rebooting Windows after installing the driver, but it give a BSoD on the next boot, which I believe is actually caused by the many many device drivers it sets up for V86 and NOT the network driver.

**Comment by copy** [comment]
Would probably good to try it with qemu with `-nic user,model=ne2k_pci`.

> Another oddity is that Window seems to PCI devices are removable?

No idea.

**Comment by BelleNottelling** [comment]
So I wanted to follow up on this.
The BSoD I was getting was unrelated to the ReactOS driver being installed, it'll happen regardless because of Vista setting up new drivers. I tried safe mode, group policy, and a few tweaks to try and prevent vista from doing that, but I had no luck. Vista would either install the driver as usual or BSoD before the initial boot.

So, I instead installed the driver in qemu, but with both it and V86 Windows doesn't show any available connections, so it seems like the driver itself isn't causing a BSoD, but it also can't make a connection. I did not even attempt the other option, as reading the page again it looks like it only works with the longhorn releases at the newest, not anything Vista+

Also, this issue of getting networking to work isn't an actual bug with V86, but instead of newer OS's not coming with drivers for the NE2K network card. However, there are serous bugs that cause issues with Vista/7 in V86 so there are still bugs.

TLDR:
 - The BSoD with Vista is a compatibility issue with V86, not the driver
 - The drive does not work at all, even in qemu
 - The other option doesn't seem to apply to the retail release of Vista

Note: I did all of my testing in Vista, but the info applies to 7 as well. In my testing they both seem to share the same behavior.

**Comment by SuperMaxusa** [comment]
I tried running ne2k_pci on Windows 7 and i have some success.

As said @BelleNottelling, Windows Vista (and 7) don't have support for the **NE2K**, but I found this [post](https://phaq.phunsites.net/2007/05/21/vista-on-xen-using-ne2000-in-favor-to-rtl8139/) about how to use this network card for Windows Vista RTM by installing the driver from Windows 2000. 

This driver also was able to install via Device Manager, but there is one problem: after each reboot, Windows installs automatically its own driver (**Ethernet Controller**), which does not work correctly with **rtl8029**

![ne2k-w7](https://user-images.githubusercontent.com/41739128/214408956-1a8800a2-e7a5-42b8-aa3c-bbb81fef2258.png)

**Comment by copy** [comment]
I'll close this issue, as everything seems to be working fine from v86's point of view.

Perhaps soon we'll get a virtio-net device, which should work in windows 7 (#1074).

**Comment by momonono17-dev** [comment]
> J'ai essayé d'exécuter ne2k_pci sur Windows 7 et j'ai eu un certain succès.
> 
> Comme dit@BelleNottellingWindows Vista (et 7) ne prennent pas en charge le **NE2K** , mais j'ai trouvé cet [article](https://phaq.phunsites.net/2007/05/21/vista-on-xen-using-ne2000-in-favor-to-rtl8139/) expliquant comment utiliser cette carte réseau pour Windows Vista RTM en installant le pilote à partir de Windows 2000.
> 
> Ce pilote a également pu être installé via le Gestionnaire de périphériques, mais il y a un problème : après chaque redémarrage, Windows installe automatiquement son propre pilote ( **Contrôleur Ethernet** ), qui ne fonctionne pas correctement avec **rtl8029**
> 
> ![ne2k-w7](https://user-images.githubusercontent.com/41739128/214408956-1a8800a2-e7a5-42b8-aa3c-bbb81fef2258.png) ![image du conducteur](https://user-images.githubusercontent.com/41739128/214424311-2ebac88d-8c30-40a7-b163-3c3d69dee152


Bonjour comment a tu mis le fichier zip sur la machine si c'est un cd fichier iso ca afficherez dvd mais je ne sais toujours pas comment a tu fait ca. windows7.img trouver sur internet archive lien du fichier utiliser windows7.iso (ne vous inquiéter pas car ce fichier est tester sur virtualbox) car les driver wifi sont officiellement ce de windows pe

**Comment by SuperMaxusa** [comment]
I added more detail instruction in #1206 (["Enabling networking" section](https://github.com/copy/v86/blob/8f97038c3e15110930e665326f49dc526fcbe287/docs/windows-nt.md#enabling-networking-ne2k))
